src/extension.ts

```diff
@@ -134,22 +33,49 @@ export function activate(context: ExtensionContext) {
 
   // Create the language client and start the client.
 
-  client = new LanguageClient(
-    "smithyLsp",
-    "Smithy LSP",
-    createServer,
-    clientOptions
-  );
-  const smithyContentProvider = createSmithyContentProvider(client);
-  context.subscriptions.push(
-    workspace.registerTextDocumentContentProvider(
-      "smithyjar",
-      smithyContentProvider
-    )
-  );
+  const version = vscode.workspace
+    .getConfiguration("smithyLsp")
+    .get("version", "`");
+
+  const lspCoordinates = vscode.workspace
+    .getConfiguration("smithyLsp")
+    .get("lspCoordinates", "`");
+
+  return getCoursierExecutable(context.globalStoragePath).then(
+    (csBinaryPath) => {
+      console.info(`Resolved coursier's binary at ${csBinaryPath}`);
+
+      const startServer = {
+        command: csBinaryPath,
+        args: [
+          "launch",
+          `${lspCoordinates}:${version}`,
+          "--ttl",
+          "1h",
+          "--",
+          "0",
+        ],
+      };
+
+      client = new LanguageClient(
+        "smithyLsp",
+        "Smithy LSP",
+        startServer,
+        clientOptions
+      );
+
+      const smithyContentProvider = createSmithyContentProvider(client);
+      context.subscriptions.push(
+        workspace.registerTextDocumentContentProvider(
+          "smithyjar",
+          smithyContentProvider
+        )
+      );
 
-  // Start the client. This will also launch the server
-  client.start();
+      // Start the client. This will also launch the server
+      client.start();
```

Comment:
FYI this is a disposable, you can pass it to `context.subscription.push`.

Reply:
oops, sorry, spoke too soon - the **client** is the disposable. I'd call `start` before `subscription.push()` and pass the client alongside the content provider.

Reply:
wow, TIL
I have just add `client.start()` as another argument to `context.subscription.push` which essentially takes a vararg